sdk/messaging/azservicebus/internal/amqpLinks.go

```diff
@@ -97,240 +101,238 @@ type CreateLinkFunc func(ctx context.Context, session AMQPSession) (AMQPSenderCl
 
 // NewAMQPLinks creates a session, starts the claim refresher and creates an associated
 // management link for a specific entity path.
-func newAMQPLinks(ns NamespaceForAMQPLinks, entityPath string, baseRetrier Retrier, createLink CreateLinkFunc) AMQPLinks {
-	l := &amqpLinks{
+func NewAMQPLinks(ns NamespaceForAMQPLinks, entityPath string, createLink CreateLinkFunc) AMQPLinks {
+	l := &AMQPLinksImpl{
 		entityPath:        entityPath,
 		managementPath:    fmt.Sprintf("%s/$management", entityPath),
 		audience:          ns.GetEntityAudience(entityPath),
 		createLink:        createLink,
-		baseRetrier:       baseRetrier,
 		closedPermanently: false,
-		revision:          1,
+		id:                LinkID{},
 		ns:                ns,
 	}
 
 	return l
 }
 
 // ManagementPath is the management path for the associated entity.
-func (links *amqpLinks) ManagementPath() string {
+func (links *AMQPLinksImpl) ManagementPath() string {
 	return links.managementPath
 }
 
 // recoverLink will recycle all associated links (mgmt, receiver, sender and session)
 // and recreate them using the link.linkCreator function.
-func (links *amqpLinks) recoverLink(ctx context.Context, theirLinkRevision *uint64) error {
+func (links *AMQPLinksImpl) recoverLink(ctx context.Context, theirLinkRevision LinkID) error {
 	ctx, span := tab.StartSpan(ctx, tracing.SpanRecoverLink)
 	defer span.End()
 
 	links.mu.RLock()
 	closedPermanently := links.closedPermanently
-	ourLinkRevision := links.revision
+	ourLinkRevision := links.id
 	links.mu.RUnlock()
 
 	if closedPermanently {
 		span.AddAttributes(tab.StringAttribute("outcome", "was_closed_permanently"))
-		return errClosedPermanently{}
+		return ErrNonRetriable{Message: "Link has been closed permanently"}
 	}
 
-	if theirLinkRevision != nil && ourLinkRevision > *theirLinkRevision {
+	// cheap check before we do the lock
+	if ourLinkRevision.Link != theirLinkRevision.Link {
```

Comment:
I'll do a naming-rename pass in another PR since this one is already big enough.

The basics are that I was previously calling this a revision, which implied some things (for instance, that you could meaningfully compare them). In reality it's just a unique ID, generated without any fanfare, and should just be treated as opaque.

Since it is an incrementing ID, however, it's useful for me for debugging purposes since it makes it simpler to tell how often something has had to be recreated.